client/my-sites/site-settings/form-general.jsx

```diff
@@ -680,33 +762,66 @@ const FormGeneral = React.createClass( {
 						{ this.syncNonPublicPostTypes() }
 
 						<CompactCard href={ '../security/' + site.slug }>
-							{ this.translate( 'View Jetpack Monitor Settings' ) }
+							{ translate( 'View Jetpack Monitor Settings' ) }
 						</CompactCard>
 						<CompactCard href={ 'https://wordpress.com/manage/' + site.ID }>
-							{ this.translate( 'Migrate followers from another WordPress.com blog' ) }
+							{ translate( 'Migrate followers from another WordPress.com blog' ) }
 						</CompactCard>
 					</div>
 					: null }
 			</div>
 		);
-	},
+	}
 
 	_showWarning( site ) {
+		const { translate } = this.props;
 		if ( ! site || ! site.options ) {
 			return;
 		}
 		if ( site.jetpack && ! site.hasMinimumJetpackVersion ) {
 			notices.warning(
-				this.translate( 'Jetpack %(version)s is required to manage Settings', {
+				translate( 'Jetpack %(version)s is required to manage Settings', {
 					args: { version: config( 'jetpack_min_version' ) }
 				} ),
 				{
-					button: this.translate( 'Update now' ),
+					button: translate( 'Update now' ),
 					href: site.options.admin_url + 'plugins.php?plugin_status=upgrade'
 				}
 			);
 		}
 	}
-} );
-
-export default protectForm( FormGeneral );
+}
+
+const connectComponent = connect(
+	state => {
+		const siteId = getSelectedSiteId( state );
+		const isRequestingSettings = isRequestingSiteSettings( state, siteId );
+		const isSavingSettings = isSavingSiteSettings( state, siteId );
+		const isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
+		const settings = getSiteSettings( state, siteId );
+		return {
+			isRequestingSettings: isRequestingSettings && ! settings,
+			isSavingSettings,
+			isSaveRequestSuccessful,
+			settings,
+			siteId
+		};
+	},
+	dispatch => {
```

Comment:
No, I don't think so, this won't have the desired behaviour, since dispatch will still be called with the cached result of `recordGoogleEvent`